### ISSUE TITLE
catboost on freebsd

### ISSUE THREAD

**Issue by nomoo** [catboost/catboost#542]
Problem: catboost on freebsd
catboost version: 0.11.1
Operating System:  FreeBSD 11.2-RELEASE

HI there! We have a bunch of things to do with catboost but our stage/prod is on freebsd, could you help with one more os ?
As far as i can see there are few mentions of bsd in the code, so seems you had supported it.

**Comment by annaveronika** [comment]
We are able to build catboost for FreeBSD, but it might be that we need to change our build a little.
For new can you please try using make files for the c++ applier (https://github.com/catboost/catboost/blob/master/make/model_interface.CLANG50-LINUX-X86_64.makefile) or of python library if you need python (https://github.com/catboost/catboost/blob/master/make/python3.CLANG50-LINUX-X86_64.makefile)?

I see that there is a lot of likes for this issue, we'll try to help with this asap!

**Comment by nomoo** [comment]
Thanks for  help!
commit: a820697
made a quick replace: 's|x86_64-linux-gnu|x86_64-unknown-freebsd11.2|' in 
error: `In file included from /wrkdirs/usr/ports/math/catboost/work/catboost-a820697/util/all_system_1.cpp:25:
/wrkdirs/usr/ports/math/catboost/work/catboost-a820697/util/system/execpath.cpp:143:34: error: invalid argument type 'TString' to unary expression
    if (FreeBSDGuessExecBasePath(~NFs::CurrentWorkingDirectory(), execPath)) {
1 error generated.
gmake[1]: *** [/wrkdirs/usr/ports/math/catboost/work/catboost-a820697/make/model_interface.CLANG50-LINUX-X86_64.makefile:2137: /wrkdirs/usr/ports/math/catboost/work/catboost-a820697/util/all_system_1.cpp.pic.o] Error 1`

**Comment by kizill** [comment]
Hello! We've fixed compilation problems on FreeBSD and i think we'll try to include freebsd binaries and wheels in some of next releases.

**Comment by nomoo** [comment]
@kizill heyas! Brilliant news, thanks!

Huh, so there will be prebuilt sources? Can i try to build everything by myself?

**Comment by Goorman** [comment]
@catboostteam Hi! Do you have any updates on this issue?

**Comment by nomoo** [comment]
@kizill, guys, we're still waiting

**Comment by kizill** [comment]
Hi all!
There is special script `ci/fixups_for_freebsd.sh` that fixes compilation problems on freebsd 11.2. Building application is trivial: 
`C_INCLUDE_PATH=/usr/local/include gmake -f make/app.CLANG50-LINUX-X86_64.makefile`
To build and install catboost wheel you can use this instruction:
```
ci/fixups_for_freebsd.sh
C_INCLUDE_PATH=/usr/local/include gmake -f make/python2.CLANG50-LINUX-X86_64.makefile
cd catboost/catboost/python-package
# to install package locally
pip install -e .
# to build catboost*.whl
python setup.py sdist bdist_wheel
```
And don't forget to use proper build parallelism vi
